spec/Section 2 -- Source Schema.md

```diff
@@ -1,244 +1,399 @@
 # Source Schema
 
+A _source schema_ is a GraphQL schema that is part of a larger _composite
+schema_. Source schemas use directives to express intent and requirements for
+the composition process. In the following chapters, we will describe the
+directives that are used to annotate a source schema.
+
 ## Directives
 
-### @entityResolver
+### @lookup
 
 ```graphql
-directive @entityResolver on FIELD_DEFINITION
+directive @lookup on FIELD_DEFINITION
 ```
 
-Entity resolvers are fields on the query root type of a subgraph that can
-resolve an entity by a stable key. The stable key is defined by the arguments of
-the field.
+The `@lookup` directive is used within a _source schema_ to specify output
+fields that can be used by the _distributed GraphQL executor_ to resolve an
+entity by a stable key.
+
+The stable key is defined by the arguments of the field. Each argument must
+match a field on the return type of the lookup field.
+
+Source schemas can provide multiple lookup fields for the same entity that
+resolve the entity by different keys.
+
+In this example, the source schema specifies that the `Product` entity can be
+resolved with the `productById` field or the `productByName` field. Both lookup
+fields are able to resolve the `Product` entity but do so with different keys.
 
 ```graphql example
-extend type Query {
-  version: Int # NOT an entity resolver.
-  personById(id: ID!): Person @entityResolver
+type Query {
```

Comment:
do we need a counter example that those two lookups could be combined into a single one?
```graphql
type Query {
  productByIdOrName(id: ID, name: String): Product @lookup
```